File: examples/laravel/app/Support/PhoneSanitizer.php
<?php

namespace App\Support;

/**
 * MyOTP requires phone numbers in international format with no leading "+"
 * or "0" -- digits only. Example: "14155551234". The API rejects anything
 * else with a 400.
 */
class PhoneSanitizer
{
    public static function sanitize(?string $value): string
    {
        $digits = preg_replace('/\D+/', '', (string) $value) ?? '';
        return ltrim($digits, '0');
    }
}

File: examples/rails/app/lib/phone_sanitizer.rb
# MyOTP requires phone numbers in international format with no leading "+"
# or "0" -- digits only. Example: "14155551234". The API rejects anything else
# with a 400.
module PhoneSanitizer
  module_function

  def call(value)
    value.to_s.gsub(/\D+/, "").sub(/\A0+/, "")
  end
end
